modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpansionModulesModel.java

```diff
@@ -86,6 +93,33 @@ public ArtifactAst expand() {
     }
 
     if (hasMacroExpansionExtension) {
+      // The macro expansion of xml sdk components causes inner flow components to be generated as a child of another component;
+      // so when generating the minimal app during lazy init, those flow components are not registered as top level components
+      // causing certain validations to fail. This code looks for those flow components and adds them as top level.
+      Stream<ComponentAst> flowComponents =
+          applicationModel.recursiveStream()
+              .filter(comp -> comp.getComponentType() != null && comp.getComponentType().equals(FLOW));
```

Comment:
be careful to only move flows from the extension that is being macroexpanded

Reply:
all extensions have been macroexpanded and added to the application model by this point